builder/swig/python.bzl

```diff
@@ -133,28 +148,63 @@ def swig_python(name, lib, shared_lib_name=None, **kwargs):
         name = swig_wrapper_name,
         shared_lib_name = shared_lib_name,
         lib = lib,
+        python_headers = python_headers,
+        libpython = select({
+            "@vaticle_dependencies//util/platform:is_linux": libpython,
+            "@vaticle_dependencies//util/platform:is_mac": None,
+            "@vaticle_dependencies//util/platform:is_windows": libpython,
+        }),
         **kwargs,
     )
 
-    def swig_cc_binary(shared_lib_filename):
-        # name doesn't accept select()
-        native.cc_binary(
-            name = shared_lib_filename,
-            deps = [lib, swig_wrapper_name],
-            srcs = [swig_wrapper_name],
-            linkshared = True,
-        )
-
-    swig_cc_binary(shared_lib_name + ".so")
-    swig_cc_binary(shared_lib_name + ".lib")
-
-    native.alias(
+    native.cc_binary(
         name = shared_lib_name,
-        actual = select({
-            "@vaticle_dependencies//util/platform:is_mac": (shared_lib_name + ".so"),
-            "@vaticle_dependencies//util/platform:is_linux": (shared_lib_name + ".so"),
-            "@vaticle_dependencies//util/platform:is_windows": (shared_lib_name + ".lib"),
-        })
+        deps = [lib, swig_wrapper_name],
+        srcs = [swig_wrapper_name],
+        linkshared = True,
+        linkopts = select({
+            "@vaticle_dependencies//util/platform:is_windows": ["ntdll.lib"],
+            "//conditions:default": [],
+        }),
+        copts = select({
+            "@vaticle_dependencies//util/platform:is_mac": ["-undefined", "dynamic_lookup"],
+            "//conditions:default": [],
+        }),
     )
 
     native.py_library(name = name, srcs = [swig_wrapper_name], data = [shared_lib_name])
+
+
+def _py_native_lib_rename_impl(ctx):
+    output_file = ctx.actions.declare_file(ctx.attr.out)
+    _copy_to_bin(ctx, ctx.files.src[0], output_file)
+    return [
+        DefaultInfo(files = depset([output_file])),
+    ]
+
+
+_py_native_lib_rename_wrapper = rule(
+    implementation = _py_native_lib_rename_impl,
+    attrs = {
+        "out": attr.string(
+            doc = "Output file name without extension",
+            mandatory = True,
+        ),
+        "src": attr.label(
+            mandatory = True,
+        ),
+    }
+)
+
+
+def py_native_lib_rename(name, out, src, visibility, **kwargs):
```

Comment:
Let's put a comment in the code here to save ourselves lots of pain in the future

Reply:
The generated dynamic library has to be copied into the output directory in order to be accessible by other rules.
We also choose the correct extension for the file.